antarest/study/storage/abstract_storage_service.py

```diff
@@ -45,12 +50,77 @@
 from antarest.study.storage.rawstudy.model.filesystem.config.files import get_playlist
 from antarest.study.storage.rawstudy.model.filesystem.config.model import Simulation
 from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy, StudyFactory
+from antarest.study.storage.rawstudy.model.filesystem.folder_node import FolderNode
+from antarest.study.storage.rawstudy.model.filesystem.inode import INode
+from antarest.study.storage.rawstudy.model.filesystem.lazy_node import LazyNode
+from antarest.study.storage.rawstudy.model.filesystem.matrix.input_series_matrix import InputSeriesMatrix
 from antarest.study.storage.rawstudy.model.helpers import FileStudyHelpers
 from antarest.study.storage.utils import extract_output_name, fix_study_root, remove_from_cache
 
 logger = logging.getLogger(__name__)
 
 
+def _assert_not_folder_node(file_node: INode[t.Any, t.Any, t.Any]) -> None:
```

Comment:
You could do this inside the get_file method before calling the extract ... to simplify the code.

Also I don't really like the name of this exception, it could be better to rename it something like:
PathIsAFolderError with a msg like f"The given path {file_node.config.path} points to a folder and not a file"

Reply:
This is basically what I was going to do but the functions `extract_xxx` should only accept a `file node`, and there is no class that abstracts this concept. 
So In the end you should raise a `folder Error` in each `extract_xxx` or create a type that excludes `non-file-nodes`.
The first option is better.